File: examples/cpp/logos_payment_service_infrastructure/simple_fraud_detection_service.cpp
#include "logos_payment_service_infrastructure/simple_fraud_detection_service.h"

namespace logos::payment::service::infrastructure {

SimpleFraudDetectionService::SimpleFraudDetectionService() {}

bool SimpleFraudDetectionService::IsFraudulent(
    int64_t amount_cents,
    const std::string& currency,
    const std::string& merchant_id) {

    // Simple rule: amounts over $5000 (500000 cents) are suspicious
    return amount_cents > 500000;
}

} // namespace logos::payment::service::infrastructure

File: examples/csharp/Logos.Payment.Service.Infrastructure.InMemory/SimpleFraudDetectionService.cs
using Logos.Payment.Service.Core.Capabilities;
using Logos.Payment.Service.Core.SharedKernel;

namespace Logos.Payment.Service.Infrastructure.InMemory;

public class SimpleFraudDetectionService : IFraudDetectionService
{
    public bool IsSuspicious(Money amount, string merchantId)
    {
        return amount.GetAmount() > 5000m;
    }
}
